src/libstd/sys/windows/stdio.rs

```diff
@@ -246,6 +246,30 @@ impl Stdout {
     pub fn new() -> io::Result<Stdout> {
         Ok(Stdout)
     }
+
+    pub fn is_tty(&self) -> bool {
+        // Because these functions do not detect if the program is being run under a Cygwin/MSYS2
+        // TTY, we just try our best to see if we are being piped on Windows Console.  If we can't
+        // tell, we fallback to assuming we are on a TTY
+        if let Ok(handle) = get_handle(c::STD_OUTPUT_HANDLE) {
+            if is_console(handle) {
+                true
+            } else {
+                // We need to check if at least one of the other stdio handles is connected to the
+                // console
+                let stdin = get_handle(c::STD_INPUT_HANDLE)
+                    .map(is_console)
+                    .unwrap_or(false);
+                let stderr = get_handle(c::STD_ERROR_HANDLE)
+                    .map(is_console)
+                    .unwrap_or(false);
+
+                !(stdin || stderr)
+            }
+        } else {
+            false
+        }
+    }
```

Comment:
I'd recommend copying the implementation for the atty crate, which I specifically fine tuned for both a standard Windows console and in MSYS2 terminals: https://github.com/softprops/atty/blob/c92ca03bd7a99c2fe8da387b32d653b81f13a667/src/lib.rs#L55

atty is used by Cargo, ripgrep, bat, fd pijul, hyperfine and anything that uses clap. So it has been pretty thoroughly vetted in the field. Can std use crates from the ecosystem? If so, maybe just using atty directly would be best. But the implementation is small enough that copying should be fine too.

Reply:
My first thought was to use `atty`, but I wasn’t sure if it’s okay to use an external crate.  I’ll copy the code for now and switch to the crate if using it is allowed.

Reply:
It is possible to depend on external crates, but it requires a lot of janky infrastructure. Unless the implementation is huge and complex, it'll probably be way easier to just copy it in.

Reply:
Okay, then I’ll just leave it as is (I copied the code in the last commit).